docs/philosophy.md

```diff
@@ -79,6 +79,11 @@ which is supported by `typing_extensions` version 4.2 and beyond makes it easier
 to test to validate the return types of functions and methods. Future work
 is intended to expand the use of `assert_type()` in the test code.
 
+After making changes to the type stubs and adding tests to verify these, please generate
+a coverage report to ensure that all changes are covered by the tests. You can generate
+the HTML report by running ``poe mypy_coverage``. The report will be generated in the
+directory ``coverage``.
```

Comment:
I'm a bit hesitant on including this paragraph.  TBH, I can't make heads or tails out of the coverage report.  It's not clear what the yellow highlighted or red highlighted code areas mean in the test files or `pyi` files.  If we're saying to people "please generate a coverage report", then we should document how to interpret that report, i.e., what do the yellow and red areas mean within each file, and if the interpretation on the tests vs. `pyi` files is different, and what are the remedies for fixing those issues (assuming that `mypy` is not buggy here!)

So you could create a new section about generating the coverage report and describe how to use it, or we could leave out the above paragraph.

You have some of that material in `docs/tests.md`, but I think you have some knowledge here that isn't obvious to others (like me!)

Reply:
Red, no types in def
Yellow, some types in def
Green, full types in def

Reply:
> Red, no types in def Yellow, some types in def Green, full types in def

When I look at something like this, I don't see how your description applies:
![image](https://user-images.githubusercontent.com/15113894/200655172-522226eb-6a74-4cc9-978b-586d76eb23ff.png)

Maybe you have to ignore the overloads?

Why is `__setitem__()` in yellow?

That's why I think extra docs on specific things to check for, and which things to ignore, would be helpful.

Reply:
I think the issue is that `Series` is generic but its type is not provided. 

Here is an example from _typing:

![image](https://user-images.githubusercontent.com/6618166/200860735-2a406be7-57a1-47bf-8c19-df4eb8f0f4ad.png)

`tuple` should have an argument but doesn't (not sure about some of the others - is `Index` generic?)


I would be fine mentioning it just in `tests.md`. If it should also be in this file, `poe mypy --coverage`.

Reply:
Yes. It would prefer `Series[Any]` over `Series`.

Reply:
I don't fully understand why `@overload` is red TBH.

Reply:
Given the discussion above,  where we don't know everything that is going on, I'd like to soften the language in the above paragraph.  I'd be more comfortable changing "please generate a coverage report" to "you may wish to generate a coverage report" and changing "to ensure that all changes" to "to understand if all changes"

Reply:
If we want to prominently push this coverage report by having it in `philosophy.md`, I wouldn't mind simply always enabling the coverage report (and not even having the option to skip it, assuming it doesn't take more time).

I don't have a strong opinion about how much to push this coverage report. For me, it seems like a similarly optional step as stubtest - in both cases, they should pass/be green, but it will take a long time.

If people try to get this report to green, we can maybe enable pyright's UnknownArgument/Type/... in the future.